Troubleshooting/heal-info-and-split-brain-resolution.md

```diff
@@ -0,0 +1,461 @@
+This command is useful for per file healing where it is known/decided that the
+file with bigger size is to be considered as source.   
+`gluster volume heal <VOLNAME> split-brain bigger-file <FILE>`  
+Here, `<FILE>` can be either the full file name as seen from the root of the volume
+(or) the gfid-string representation of the file, which sometimes gets displayed
+in the heal info command's output. Once this command is executed, the replica containing the <FILE> with bigger
+size is found out and heal is completed with that bricks as a source.
+
+### Example :
+Consider the earlier output of heal info split-brain command.
+
+Before healing the file, notice file size and md5 checksums :  
+~~~
+On brick b1:
+# stat b1/dir/file1 
+  File: ‘b1/dir/file1’
+  Size: 17              Blocks: 16         IO Block: 4096   regular file
+Device: fd03h/64771d    Inode: 919362      Links: 2
+Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
+Access: 2015-03-06 13:55:40.149897333 +0530
+Modify: 2015-03-06 13:55:37.206880347 +0530
+Change: 2015-03-06 13:55:37.206880347 +0530
+ Birth: -
+
+# md5sum b1/dir/file1 
+040751929ceabf77c3c0b3b662f341a8  b1/dir/file1
+
+On brick b2:
+# stat b2/dir/file1 
+  File: ‘b2/dir/file1’
+  Size: 13              Blocks: 16         IO Block: 4096   regular file
+Device: fd03h/64771d    Inode: 919365      Links: 2
+Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
+Access: 2015-03-06 13:54:22.974451898 +0530
+Modify: 2015-03-06 13:52:22.910758923 +0530
+Change: 2015-03-06 13:52:22.910758923 +0530
+ Birth: -
+# md5sum b2/dir/file1 
+cb11635a45d45668a403145059c2a0d5  b2/dir/file1
+~~~
+**Healing file1 using the above command** :-    
+`gluster volume heal test split-brain bigger-file /dir/file1`  
+Healed /dir/file1.
+
+After healing is complete, the md5sum and file size on both bricks should be the same.
+~~~
+On brick b1:
+# stat b1/dir/file1 
+  File: ‘b1/dir/file1’
+  Size: 17              Blocks: 16         IO Block: 4096   regular file
+Device: fd03h/64771d    Inode: 919362      Links: 2
+Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
+Access: 2015-03-06 14:17:27.752429505 +0530
+Modify: 2015-03-06 13:55:37.206880347 +0530
+Change: 2015-03-06 14:17:12.880343950 +0530
+ Birth: -
+# md5sum b1/dir/file1 
+040751929ceabf77c3c0b3b662f341a8  b1/dir/file1
+
+On brick b2:
+# stat b2/dir/file1 
+  File: ‘b2/dir/file1’
+  Size: 17              Blocks: 16         IO Block: 4096   regular file
+Device: fd03h/64771d    Inode: 919365      Links: 2
+Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
+Access: 2015-03-06 14:17:23.249403600 +0530
+Modify: 2015-03-06 13:55:37.206880000 +0530
+Change: 2015-03-06 14:17:12.881343955 +0530
+ Birth: -
+
+# md5sum b2/dir/file1 
+040751929ceabf77c3c0b3b662f341a8  b2/dir/file1
+~~~
+## ii) Select the file with the latest mtime as source  
+`gluster volume heal <VOLNAME> split-brain latest-mtime <FILE>`  
```

Comment:
Nope, both 'bigger-file' and 'mtime' policies are on a per-file basis, so you would have to run the command for every file that is in data or metadata split-brain.